src/Plugin/Block/FooterBlockBase.php

```diff
@@ -90,18 +101,49 @@ protected function setSiteSpecificFooter(array &$build, CacheableMetadata &$cach
   /**
    * Retrieve a list of custom footer link configuration entities.
    *
-   * @param string $type
-   *   Type of configs.
    * @param \Drupal\Core\Cache\CacheableMetadata $cache
    *   The Cacheable metadata abject.
    *
    * @return array
    *   The array of links.
    */
-  protected function getSiteSpecificFooterLinks(string $type, CacheableMetadata &$cache): array {
+  protected function getGeneralFooterLinks(CacheableMetadata &$cache): array {
+    $cache->addCacheTags($this->linkManager->getListCacheTags());
```

Comment:
Simple add entity type cache tags from the different storages here by calling `addCacheTags` twice, as done below in getSocialFooterLinks().